src/index.ts

```diff
@@ -509,17 +487,98 @@ const initialize = (apiKey: string, target: Target, options?: Options): Result =
       }
     }
 
-    eventSource.addEventListener('*', (msg: any) => {
-      const event: StreamEvent = JSON.parse(msg.data)
+    const processData = (data: any): void => {
+      const lines = data.toString().split(/\r?\n/);
+      lines.forEach((line) => processLine(line));
+    }
 
-      logDebug('Received event from stream: ', event)
+    const processLine = (line: string): void => {
+      if (line.startsWith('data:')) {
+        const event: StreamEvent = JSON.parse(line.substring(5))
+        logDebug('Received event from stream: ', event)
 
-      if (event.domain === 'flag') {
-        handleFlagEvent(event)
-      } else if (event.domain === 'target-segment') {
-        handleSegmentEvent(event)
+        if (event.domain === 'flag') {
+          handleFlagEvent(event)
+        } else if (event.domain === 'target-segment') {
+          handleSegmentEvent(event)
+        }
       }
-    })
+    }
+
+    const getRandom = (min, max) => {
+      return Math.round(Math.random() * (max - min) + min)
+    }
+
+    const onConnected = () => {
+      logDebug('Stream connected')
+      eventBus.emit(Event.CONNECTED)
+    };
+
+    const onDisconnect = () => {
+      clearInterval(readTimeoutCheckerId)
+      const reconnectDelayMs = getRandom(1000, 10000)
+      logDebug('Stream disconnected, will reconnect in ' + reconnectDelayMs + 'ms')
+      eventBus.emit(Event.DISCONNECTED)
+      setTimeout(() => startStream(), reconnectDelayMs)
+    };
+
+    const onFailed = (msg: string) => {
+      if (msg != null) {
```

Comment:
Might as well do a `truthy` check instead of just looking for not being `null`
```suggestion
      if (!!msg) {
```